rpcserver.go

```diff
@@ -3060,18 +3060,39 @@ func SignRawTransaction(w *wallet.Wallet, chainSvr *chain.Client, icmd interface
 			// Failure to sign isn't an error, it just means that
 			// the tx isn't complete.
 			if err != nil {
-				complete = false
+				signErrors = append(signErrors,
+					btcjson.SignRawTransactionError{
+						TxID:      txIn.PreviousOutPoint.Hash.String(),
+						Vout:      txIn.PreviousOutPoint.Index,
+						ScriptSig: hex.EncodeToString(txIn.SignatureScript),
+						Sequence:  txIn.Sequence,
+						Error:     err.Error(),
+					})
 				continue
 			}
 			txIn.SignatureScript = script
 		}
 
+		var tmpErr error
 		// Either it was already signed or we just signed it.
 		// Find out if it is completely satisfied or still needs more.
 		vm, err := txscript.NewEngine(input, msgTx, i,
 			txscript.StandardVerifyFlags)
-		if err != nil || vm.Execute() != nil {
-			complete = false
+
+		if err != nil {
+			tmpErr = err
+		} else if vm.Execute() != nil {
+			tmpErr = err
```

Comment:
This doesn't look right.  Did you mean to set tmpErr to the result of vm.Execute?